### ISSUE TITLE
AZ Token not refreshed when switching tenants

### ISSUE THREAD

**Issue by dduede** [rmbolger/Posh-ACME#498]
I guess this can be considered a bug.

I'm the administrator of several azure AD tenants, each with their own set of subscriptions and DNS Zones.
I can order a certificate after logging into azure just fine, for as many domains as I want - as long as they are in the same azure tenant.

If I switch to a different azure tenant, the whole process breaks. I keep getting the azure error `Access token is from the wrong issuer` - which makes no sense. I just requested a new AccessToken from the correct tenant.

After some debugging via `-verbose` and `-debug` I noticed the following statement:
```
DEBUG: Current Token Expires: 05/25/2023 17:39:57 +00:00
DEBUG: Existing token has not expired.
```
To me, this didn't make any sense because the token expiry didn't match the one I passed in - so it looks like the script is caching the token (is it `$script:AZToken`?) and only grabs something new, if the token expired.

The problem is, this issue also happens with the Service Principal login and most likely with any other authentication method as well: If there was a successful authentication, the token will be reused unless its expired - even though the new credentials point to a different subscription and tenant.

## Reproduction steps
Posh-ACME Version 4.17.1
```powershell
# Login into tenant
Login-AzAccount -Tenant my-tenant-one.onmicrosoft.com

$pArgs = @{
    AZSubscriptionId = (Get-AzContext).Subscription.Id
    AZAccessToken = (Get-AzAccessToken).Token
}

# Request Certificate
New-PACertificate -Domain xxx -Plugin Azure -PluginArgs $pArgs # this is successful

/* handling new certificate here */

# Login into another tenant
Login-AzAccount -Tenant my-tenant-two.onmicrosoft.com

$pArgs = @{
    AZSubscriptionId = (Get-AzContext).Subscription.Id
    AZAccessToken = (Get-AzAccessToken).Token
}

# Request Certificate
New-PACertificate -Domain yyy -Plugin Azure -PluginArgs $pArgs # this fails with "InvalidAuthenticationTokenTenant"
```

## Expected Result
Both certificate requests are successful

## Actual Result
Authentication error
```
{"error":{"code":"InvalidAuthenticationTokenTenant","message":"The access token is from the wrong issuer 'https://sts.windows.net/[my-tenant-one]/'. It must match one of the tenants 'https://sts.windows.net/[my-main-tenant]/,https://sts.windows.net/[my-tenant-one]/' associated with this subscription. Please use any authority (URL) from 'https://login.windows.net/[my-main-tenant],https://login.windows.net/[my-tenant-one]' to get the token. Note, if the subscription is transferred to another tenant there is no impact to the services, but information about new tenant could take time to propagate (up to an hour). If you just transferred your subscription and see this error message, please try back later."}}
```

## Workaround
Restarting the powershell session clears the cached values, causing a new authentication with the correct tenant.

**Comment by rmbolger** [comment]
Hey @dduede, thanks for reaching out. Yep, that definitely looks like an unintended consequence of the current token caching implementation. I'll have to figure out a way to recognize the context change and invalidate the current token if that happens.

Until then, should also be able to workaround the issue by re-importing the module with the `-Force` flag which should wipe out any saved module-scoped variables.

```powershell
Import-Module Posh-ACME -Force
```

**Comment by rmbolger** [comment]
I committed a change that should fix the issue. But it broke a bunch of Pester tests for the Azure plugin and I haven't had time to fix them yet. But if you wanted to test the dev copy in the main branch, it should work.

**Comment by rmbolger** [comment]
Fix is now live in 4.18.0
